app/src/main/java/com/sethpthomas91/httpserver/ServerSocketWrapper.java

```diff
@@ -0,0 +1,76 @@
+package com.sethpthomas91.httpserver;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.text.Format;
+
+public class ServerSocketWrapper implements ServerSocketWrapperInterface{
+    ServerSocket serverSocket = null;
+    Socket clientSocket = null;
+    BufferedReader clientReader = null;
+    PrintWriter clientWriter = null;
+    boolean isListening = false;
+    int port;
+
+    public ServerSocketWrapper(int newPort){
+        this.port = newPort;
+    }
+
+    private void createWriter() throws IOException {
+        clientWriter = new PrintWriter(clientSocket.getOutputStream(), true);
+        System.out.println("[Client Writer Created]");
+    }
+
+    private void createReader() throws IOException {
+        clientReader = new BufferedReader( new InputStreamReader(clientSocket.getInputStream()));
+        System.out.println("[Client Reader Created]");
+    }
+
+    private void createClientSocket() {
+        try {
+            clientSocket = serverSocket.accept();
+            System.out.println("[Connected to Client]");
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.out.println("[Failed to Create Client]");
+        }
+    }
+
+    @Override
+    public void createServerSocket(int port) throws IOException {
+        serverSocket = new ServerSocket(port);
+        System.out.println(String.format("[LISTENING ON PORT: %s]", port));
+    }
+
+    @Override
+    public void listen() throws IOException {
+        isListening = true;
+        createServerSocket(this.port);
+        while (isListening) {
+            createClientSocket();
+            createReader();
+            createWriter();
+            displayIncomingMessage();
+        }
+    }
+
+    public void disconnect() throws IOException {
+        System.out.println(String.format("[Client Socket at %s Disconnected]", clientSocket.getPort()));
+        clientSocket.close();
+    }
+
+    public void displayIncomingMessage() throws IOException {
+        System.out.println(clientReader.readLine());
+        sendHardCodedMessage();
+        disconnect();
+    }
+    public void sendHardCodedMessage() {
+        String CRLF = "\r\n";
+        String newMessage = "HTTP/1.1 200 OK" + CRLF;
+        clientWriter.println(newMessage);
+    }
```

Comment:
Lots of `public` methods and no test for this class. We tested a lot through the ServerRunner, but we still want test of public methods when possible, maybe you can pull some of those test down a level or clean up the API of this class so there's less logic accessible publicly.

Reply:
I did end up fixing these public methods a bit in the next iteration. I have 3 public methods now, and the rest are private. Agreed on the testing. I still don't know how I want to better handle processing the requests yet. I was hoping that it would illuminate in my TDD, but not just yet.

Reply:
Yeah, I think you're on the right path, the TDD will start doing that as you move through the requirements. Sounds good.